[CreateMeshFromMask] clean memory management, avoid crash after process macOS14

mathildemerle opened the pull request:
This PR aims to solves problems leading to the crash of CreateMeshFromMask on some macOS https://github.com/Inria-Asclepios/medInria-public/issues/886

At the end of the process call in `medCreateMeshFromMaskToolBox::run()` the process was automatically destroyed whereas it's still used in the toolbox in a method called in case of process success. Depending of the timing on some machines/modes/OS we could have a crash. The PR cleans also the way the process is stored, and variables inside the process handled.

This PR is needed, and we'll test the binaries on different machines.

:m:

mathildemerle committed:
[CreateMeshFromMask] clean memory management, avoid crash after process macOS14

fcollot commented:
Changing the memory management of a core class that is used by many other classes is not a good idea.
The bug is caused by the toolbox storing the process as a raw pointer instead of a `dtkSmartPointer`. Making it a `dtkSmartPointer` (like is done in all the other classes using `medRunnableProcess`) will fix it.

mathildemerle committed:
[CreateMesh] change to smart pointers

mathildemerle commented:
i tried at the beginning but there was an other problem that's why i switched back, but indeed it's better, and also i updated with the same change https://github.com/Inria-Asclepios/music/pull/1108

fcollot commented on src/plugins/legacy/medCreateMeshFromMask/medCreateMeshFromMaskToolBox.cpp:
```diff
@@ -123,6 +122,7 @@ medCreateMeshFromMaskToolBox::medCreateMeshFromMaskToolBox(QWidget *parent)
     this->addWidget(widget);
 
     d->view = nullptr;
+    d->process = nullptr;
```
Smart pointers do not need to be initialized.

mathildemerle committed:
Update medCreateMeshFromMaskToolBox.cpp -- rm process to nullptr
